.gitignore

```diff
@@ -24,4 +24,256 @@ _dotTrace*
 # rd-gen output
 *.Generated.cs
 *.Generated.kt
-*.bak
\ No newline at end of file
+*.bak
+
+
+## Ignore Visual Studio temporary files, build results, and
+## files generated by popular Visual Studio add-ons.
+
+# User-specific files
+*.suo
+*.user
+*.userosscache
+*.sln.docstates
+
+# User-specific files (MonoDevelop/Xamarin Studio)
+*.userprefs
+
+# Build results
+[Dd]ebug/
+[Dd]ebugPublic/
+[Rr]elease/
+[Rr]eleases/
+[Xx]64/
+[Xx]86/
+[Bb]uild/
+bld/
+[Bb]in/
+[Oo]bj/
+
+# Visual Studio 2015 cache/options directory
+.vs/
+
+dist/
+
+**/wwwroot/sw.js
+**/wwwroot/assets/slides/
+**/wwwroot/assets/qabs/
+**/wwwroot/assets/layout/
+
+#Idea files
+.idea/
```

Comment:
It should not filter out the whole `.idea` folder because it contains run configuration visible from Intellij IDEA.